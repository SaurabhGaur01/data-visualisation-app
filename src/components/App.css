.table-content {
  min-width: 750px;
}

.card-container {
  width: 600px;
}

.filter-by {
  float: right;
}

.filter-heading {
  font-weight: 800;
}

.chart-container {
  width: 100%;
  height: 410px;
}

thead > tr {
  background: antiquewhite;
}

tr:nth-child(even) {
  background: #ccc;
}
