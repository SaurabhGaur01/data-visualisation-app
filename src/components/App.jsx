import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import BoatRamps from './Content/BoatRamps/BoatRamps';
import ConstructionMaterial from './Content/Filter/ConstructionMaterial';
import Filter from './Content/Filter/Filter';
import handleGetData from '../thunks/handleGetData';
import './App.css';

const App = ({
  actionHandleGetData,
}) =>{
  /* 
    Below action will be dispatched before component renders and make an API call,
    store the response in redux store
    */
    React.useEffect(() => {
      actionHandleGetData();
  }, [actionHandleGetData]);

  return (
    <div className="dashboard-content">
      <h5 className="text-center">
        BoatRamp Dashboard
      </h5>
      <Filter />
      <Card className="card-container mt-4 ml-3 mx-auto" variant="outlined">
        <CardContent>
          <ConstructionMaterial />
        </CardContent>  
      </Card>
      <BoatRamps />
    </div> 
  )
}  

App.propTypes = {
  actionHandleGetData: PropTypes.func.isRequired,
};

// Getting action from redux store for dispatching 
const mapDispatchAsProps = {
  actionHandleGetData: handleGetData,
}

const hocChain = compose(
  connect(null, mapDispatchAsProps),
);

export { App as TestableApp };

export default hocChain(App);
