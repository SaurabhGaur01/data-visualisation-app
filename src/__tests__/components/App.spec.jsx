import React from 'react';
import { shallow } from 'enzyme';
import { TestableApp } from '../../components/App';

describe('App Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionHandleGetData: jest.fn(),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableApp {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('useEffect', () => {
        it('should call actionHandleGetData when component mounts', () => {
            jest.spyOn(React, 'useEffect').mockImplementationOnce(f => f());
            shallow(<TestableApp {...props} />);
            expect(props.actionHandleGetData).toHaveBeenCalledTimes(1);
        });
    })
})