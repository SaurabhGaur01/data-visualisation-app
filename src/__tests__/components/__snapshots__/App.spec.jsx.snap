// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
<div
  className="dashboard-content"
>
  <h5
    className="text-center"
  >
    BoatRamp Dashboard
  </h5>
  <Connect(Filter) />
  <WithStyles(ForwardRef(Card))
    className="card-container mt-4 ml-3 mx-auto"
    variant="outlined"
  >
    <WithStyles(ForwardRef(CardContent))>
      <Connect(ConstructionMaterial) />
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
  <Connect(BoatRamps) />
</div>
`;
